Projects/App/Sources/Screens/ChatScreen/Reusables/Message+Reusable.swift

```diff
@@ -95,6 +97,27 @@ extension Message: Reusable {
 
             return 84 + imageHeight + size.height + largerMarginTop + extraHeightForTimeStampLabel
         }
+        if case .deepLink = type {
+            let title = "Go to contract"
+            let attributedString = NSAttributedString(
+                styledText: StyledText(text: title, style: UIColor.brandStyle(.chatMessage))
+            )
+            let size = attributedString.boundingRect(
+                with: CGSize(width: 227.77, height: CGFloat(Int.max)),
+                options: [.usesLineFragmentOrigin, .usesFontLeading],
+                context: nil
+            )
+            let imageHeight: CGFloat = {
+                let imageToShow = hCoreUIAssets.pillowHome.image
+                let ratio = imageToShow.size.width / imageToShow.size.height
+                return (227.77) / ratio
+            }()
+            if isRelatedToPreviousMessage {
+                return imageHeight + smallerMarginTop + extraHeightForTimeStampLabel
+            }
+
+            return imageHeight + largerMarginTop + extraHeightForTimeStampLabel
```

Comment:
height is not calculated right. We aren't taking title height in consideration and we are probably wrongly calculating imageHeight  - so it looks good now - and missed fixed size for spacing.